eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php

```diff
@@ -0,0 +1,142 @@
+        $this->checkDatabase();
+
+        $type = $input->getArgument( 'type' );
+        $installer = $this->getInstaller( $type );
+        if ( $installer === false )
+        {
+                $output->writeln( "Unknown install type '$type'" );
+                exit( self::EXIT_UNKNOWN_INSTALL_TYPE );
+        }
+
+        $installer->setOutput( $output );
+
+        $installer->createConfiguration();
+        $installer->importSchema();
+        $installer->importData();
+        $installer->importBinaries();
+    }
+
+    private function checkPermissions()
+    {
+        if ( !is_writable( 'ezpublish/config' ) )
+        {
+            $this->output->writeln( "ezpublish/config is not writable" );
+            exit( self::EXIT_MISSING_PERMISSIONS );
+        }
+    }
+
+    private function checkParameters()
+    {
+        $parametersFile = 'ezpublish/config/parameters.yml';
+        if ( !is_file( $parametersFile ) )
+        {
+            $this->output->writeln( "Required configuration file $parametersFile not found" );
+            exit( self::EXIT_PARAMETERS_NOT_FOUND );
+        }
+    }
+
+    /**
+     * @throws \Exception if an unexpected database error occurs
+     */
+    private function configuredDatabaseExists()
+    {
+        $this->db = $this->getContainer()->get( 'database_connection' );
+        try
+        {
+            $this->db->connect();
+        }
+        catch ( ConnectionException $e )
+        {
+            // @todo 1049 is MySQL's code for "database doesn't exist", refactor
+            if ( $e->getPrevious()->getCode() == 1049 )
+            {
+                return false;
+            }
+            throw $e;
+        }
+        return true;
+    }
+
+    private function checkDatabase()
+    {
+        try
+        {
+            if ( !$this->configuredDatabaseExists() )
+            {
+                $this->output->writeln(
+                    sprintf(
+                        "The configured database '%s' does not exist",
+                        $this->db->getDatabase()
+                    )
+                );
+                exit( self::EXIT_DATABASE_NOT_FOUND_ERROR );
+            }
+        }
+        catch ( ConnectionException $e )
+        {
+            $this->output->writeln( "An error occured connecting to the database:" );
+            $this->output->writeln( $e->getMessage() );
+            $this->output->writeln( "Please check the database configuration in parameters.yml" );
+            exit( self::EXIT_GENERAL_DATABASE_ERROR );
+        }
+    }
+
+    /**
+     * @param $type
+     *
+     * @return \EzSystems\PlatformInstallerBundle\Installer\Installer
+     */
+    private function getInstaller( $type )
+    {
+        $installers = $this->getContainer()->getParameter( 'ezplatform.installers' );
+        if ( !isset( $installers[$type] ) )
+        {
+            return false;
+        }
+
+        return $this->getContainer()->get( $installers[$type] );
```

Comment:
Considering you modify this command to be a service, you could add installers with a setter (e.g. `addInstaller()`) in your compiler pass.
You may keep `ezplatform.installers` if needed.